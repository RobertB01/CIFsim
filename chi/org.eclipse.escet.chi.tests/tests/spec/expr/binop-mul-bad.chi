################################################################################
# Copyright (c) 2010, 2023 Contributors to the Eclipse Foundation
#
# See the NOTICE file(s) distributed with this work for additional
# information regarding copyright ownership.
#
# This program and the accompanying materials are made available under the terms
# of the MIT License which is available at https://opensource.org/licenses/MIT
#
# SPDX-License-Identifier: MIT
#################################################################################

proc SP1(chan? int a;  chan?  int b): set chan? int x = {a} * {b}; pass end
proc SP2(chan! int a;  chan?  int b): set chan? int x = {a} * {b}; pass end
proc SP3(chan!? int a; chan?  int b): set chan? int x = {a} * {b}; pass end
proc SP4(chan? int a;  chan!  int b): set chan? int x = {a} * {b}; pass end
proc SP5(chan! int a;  chan!  int b): set chan? int x = {a} * {b}; pass end
proc SP6(chan!? int a; chan!  int b): set chan? int x = {a} * {b}; pass end
proc SP7(chan? int a;  chan!? int b): set chan? int x = {a} * {b}; pass end
proc SP8(chan! int a;  chan!? int b): set chan? int x = {a} * {b}; pass end
proc SP9(chan!? int a; chan!? int b): set chan? int x = {a} * {b}; pass end

proc SQ1(chan? int a;  chan?  int b): set chan! int x = {a} * {b}; pass end
proc SQ2(chan! int a;  chan?  int b): set chan! int x = {a} * {b}; pass end
proc SQ3(chan!? int a; chan?  int b): set chan! int x = {a} * {b}; pass end
proc SQ4(chan? int a;  chan!  int b): set chan! int x = {a} * {b}; pass end
proc SQ5(chan! int a;  chan!  int b): set chan! int x = {a} * {b}; pass end
proc SQ6(chan!? int a; chan!  int b): set chan! int x = {a} * {b}; pass end
proc SQ7(chan? int a;  chan!? int b): set chan! int x = {a} * {b}; pass end
proc SQ8(chan! int a;  chan!? int b): set chan! int x = {a} * {b}; pass end
proc SQ9(chan!? int a; chan!? int b): set chan! int x = {a} * {b}; pass end

proc SR1(chan? int a;  chan?  int b): set chan!? int x = {a} * {b}; pass end
proc SR2(chan! int a;  chan?  int b): set chan!? int x = {a} * {b}; pass end
proc SR3(chan!? int a; chan?  int b): set chan!? int x = {a} * {b}; pass end
proc SR4(chan? int a;  chan!  int b): set chan!? int x = {a} * {b}; pass end
proc SR5(chan! int a;  chan!  int b): set chan!? int x = {a} * {b}; pass end
proc SR6(chan!? int a; chan!  int b): set chan!? int x = {a} * {b}; pass end
proc SR7(chan? int a;  chan!? int b): set chan!? int x = {a} * {b}; pass end
proc SR8(chan! int a;  chan!? int b): set chan!? int x = {a} * {b}; pass end
proc SR9(chan!? int a; chan!? int b): set chan!? int x = {a} * {b}; pass end
