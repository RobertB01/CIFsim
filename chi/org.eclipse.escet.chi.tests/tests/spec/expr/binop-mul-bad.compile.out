Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 14, column 61: Incorrectly typed multiplication (cannot multiply types "set chan! int" and "set chan? int").
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 16, column 61: Incorrectly typed multiplication (cannot multiply types "set chan? int" and "set chan! int").
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 17, column 53: Cannot assign value with type "set chan! int" to variables with type "set chan? int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 18, column 53: Cannot assign value with type "set chan! int" to variables with type "set chan? int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 20, column 53: Cannot assign value with type "set chan! int" to variables with type "set chan? int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 23, column 53: Cannot assign value with type "set chan? int" to variables with type "set chan! int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 24, column 61: Incorrectly typed multiplication (cannot multiply types "set chan! int" and "set chan? int").
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 25, column 53: Cannot assign value with type "set chan? int" to variables with type "set chan! int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 26, column 61: Incorrectly typed multiplication (cannot multiply types "set chan? int" and "set chan! int").
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 29, column 53: Cannot assign value with type "set chan? int" to variables with type "set chan! int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 33, column 54: Cannot assign value with type "set chan? int" to variables with type "set chan!? int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 34, column 62: Incorrectly typed multiplication (cannot multiply types "set chan! int" and "set chan? int").
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 35, column 54: Cannot assign value with type "set chan? int" to variables with type "set chan!? int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 36, column 62: Incorrectly typed multiplication (cannot multiply types "set chan? int" and "set chan! int").
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 37, column 54: Cannot assign value with type "set chan! int" to variables with type "set chan!? int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 38, column 54: Cannot assign value with type "set chan! int" to variables with type "set chan!? int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 39, column 54: Cannot assign value with type "set chan? int" to variables with type "set chan!? int".
Error: File "spec/expr/binop-mul-bad.chi": Semantic error at line 40, column 54: Cannot assign value with type "set chan! int" to variables with type "set chan!? int".
